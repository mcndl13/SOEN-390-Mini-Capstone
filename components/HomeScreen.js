// HomeScreen.js
import React from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from "react-native";

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Welcome</Text>
      <ScrollView>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CampusMap")}
        >
          <Text style={styles.buttonText}>Explore Campus Map</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("Directions")}
        >
          <Text style={styles.buttonText}>Get Directions</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CalendarIntegration")}
        >
          <Text style={styles.buttonText}>Connect to Google Calendar</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("IndoorDirections")}
        >
          <Text style={styles.buttonText}>Indoor Navigation</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("PointsOfInterest")}
        >
          <Text style={styles.buttonText}>Find Points of Interest</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  title: {
    fontSize: 32,
    fontWeight: "semi-bold",
    marginTop: 60,
    marginBottom: 40,
  },
  button: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#912338",
    padding: 20,
    margin: 5,
    borderRadius: 30,
  },
  buttonText: {
    color: "white",
    fontSize: 18,
  },
});
