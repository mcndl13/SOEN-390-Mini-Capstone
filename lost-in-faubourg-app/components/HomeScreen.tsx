import React, { useEffect, useState } from "react";
import { Linking, View, Text, StyleSheet, TouchableOpacity, ScrollView, Alert } from "react-native";
import * as Location from "expo-location";
import { NavigationProp } from "@react-navigation/native";

type HomeScreenProps = {
  navigation: NavigationProp<any>;
};

const HomeScreen = ({ navigation }: HomeScreenProps) => {
  const [hasPermission, setHasPermission] = useState(false);

  const requestLocationPermission = async () => {
    const { status } = await Location.requestForegroundPermissionsAsync();
    if (status !== "granted") {
      Alert.alert("Permission Denied", "Allow location access to use maps.");
      return false;
    }
    return true;
  };

  useEffect(() => {
    const getPermission = async () => {
      const permission = await requestLocationPermission();
      setHasPermission(permission);
    };
    getPermission();
  }, []);

  // Function to open the shuttle schedule webpage
  const openShuttleSchedule = async () => {
    const url = "https://www.concordia.ca/maps/shuttle-bus.html#depart";
    const supported = await Linking.canOpenURL(url);
    
    if (supported) {
      await Linking.openURL(url);
    } else {
      console.error("Cannot open URL: " + url);
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Welcome</Text>
      <ScrollView>
        <TouchableOpacity
          style={styles.button}
          onPress={() => {
            if (hasPermission) {
              navigation.navigate("CampusMap");
            } else {
              Alert.alert("Permission Needed", "You need to allow location access.");
            }
          }}
        >
          <Text style={styles.buttonText}>Explore Campus Map</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("Directions")}
        >
          <Text style={styles.buttonText}>Get Directions</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("CalendarIntegration")}
        >
          <Text style={styles.buttonText}>Connect to Google Calendar</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("IndoorDirections")}
        >
          <Text style={styles.buttonText}>Indoor Navigation</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={() => navigation.navigate("PointsOfInterest")}
        >
          <Text style={styles.buttonText}>Find Points of Interest</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.button}
          onPress={openShuttleSchedule}
        >
          <Text style={styles.buttonText}>Shuttle Schedule</Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  title: {
    fontSize: 32,
    fontWeight: "600",
    marginTop: 60,
    marginBottom: 40,
  },
  button: {
    backgroundColor: "#912338",
    padding: 20,
    margin: 5,
    borderRadius: 30,
    justifyContent: "center",
    alignItems: "center",
  },
  buttonText: {
    color: "white",
    fontSize: 18,
  },
});

export default HomeScreen;