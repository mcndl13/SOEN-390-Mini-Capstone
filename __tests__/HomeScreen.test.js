import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreen from '../components/HomeScreen';

const mockNavigation = { navigate: jest.fn() };

test('HomeScreen renders correctly', () => {
  const tree = renderer.create(<HomeScreen navigation={mockNavigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
