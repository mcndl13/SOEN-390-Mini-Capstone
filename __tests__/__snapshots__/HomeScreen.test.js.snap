// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Welcome
  </Text>
  <RCTScrollView>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#007BFF",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Explore Campus Map
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#007BFF",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Get Directions
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#007BFF",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Connect to Google Calendar
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#007BFF",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Indoor Navigation
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#007BFF",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Find Points of Interest
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
